client.go

```diff
@@ -63,118 +60,26 @@ type Client struct {
-		AccessKey: accessKey,
-		HTTPClient: &http.Client{
-			Timeout: httpClientTimeout,
-		},
-		request: requestV2,
-	}
 }
 
 // New creates a new MessageBird client object.
-// If the request to MessageBird API will return
-// and error, it will simply return ErrResponse.
-// Deprecated: use NewV2 until v2 release.
 func New(accessKey string) *Client {
 	return &Client{
 		AccessKey: accessKey,
 		HTTPClient: &http.Client{
 			Timeout: httpClientTimeout,
 		},
-		request: requestV1,
 	}
 }
 
-// Request is for internal use only and unstable.
-func (c *Client) Request(v interface{}, method, path string, data interface{}) error {
-	return c.request(c, v, method, path, data)
-}
-
-func requestV1(c *Client, v interface{}, method, path string, data interface{}) error {
-	if !strings.HasPrefix(path, "https://") && !strings.HasPrefix(path, "http://") {
-		path = fmt.Sprintf("%s/%s", Endpoint, path)
-	}
-	uri, err := url.Parse(path)
-	if err != nil {
-		return err
-	}
-
-	var jsonEncoded []byte
-	if data != nil {
-		jsonEncoded, err = json.Marshal(data)
-		if err != nil {
-			return err
-		}
-	}
-
-	request, err := http.NewRequest(method, uri.String(), bytes.NewBuffer(jsonEncoded))
-	if err != nil {
-		return err
-	}
-
-	request.Header.Set("Content-Type", "application/json")
-	request.Header.Set("Accept", "application/json")
-	request.Header.Set("Authorization", "AccessKey "+c.AccessKey)
-	request.Header.Set("User-Agent", "MessageBird/ApiClient/"+ClientVersion+" Go/"+runtime.Version())
-
-	if c.DebugLog != nil {
-		if data != nil {
-			c.DebugLog.Printf("HTTP REQUEST: %s %s %s", method, uri.String(), jsonEncoded)
-		} else {
-			c.DebugLog.Printf("HTTP REQUEST: %s %s", method, uri.String())
-		}
-	}
-
-	response, err := c.HTTPClient.Do(request)
-	if err != nil {
-		return err
-	}
-
-	defer response.Body.Close()
-
-	responseBody, err := ioutil.ReadAll(response.Body)
-	if err != nil {
-		return err
-	}
-
-	if c.DebugLog != nil {
-		c.DebugLog.Printf("HTTP RESPONSE: %s", string(responseBody))
-	}
-
-	// Status code 500 is a server error and means nothing can be done at this
-	// point.
-	if response.StatusCode == 500 {
-		return ErrUnexpectedResponse
-	}
-	// Status codes 200 and 201 are indicative of being able to convert the
-	// response body to the struct that was specified.
-	if response.StatusCode == 200 || response.StatusCode == 201 {
-		if err := json.Unmarshal(responseBody, &v); err != nil {
-			return fmt.Errorf("could not decode response JSON, %s: %v", string(responseBody), err)
-		}
-		return nil
-	}
-
-	// We're dealing with an API error here. try to decode it, but don't do
-	// anything with the error. This is because not all values of `v` have
-	// `Error` properties and decoding could fail.
-	json.Unmarshal(responseBody, &v)
-
-	// Anything else than a 200/201/500 should be a JSON error.
-	return ErrResponse
+// NewV2 creates a new MessageBird client object.
+// Deprecated: use New factory instead.
+func NewV2(accessKey string) *Client {
```

Comment:
Please remove `NewV2` - no need for deprecating it as we're releasing this with a major version bump